### PR TITLE
Unfreeze pymongo and add integration test with MongoDB v7

### PR DESCRIPTION
**Pull request by northwestwitch**
Changes introduced by this PR:
- Unfreeze pymongo requirement and removed deprecated pymongo code to support latest version of MongoDB
- Added Integration test with MongoDB v7 

**How to test**:
- Make sure the new integration test action works

**Review:**
- [x] code approved by HS
- [x] tests executed by GitHub actions

Minor release

### CODE REVIEW

**Comment by northwestwitch** [comment]
@henrikstranneheim I have another easy review when you have time, thanks!

File: .github/workflows/integration_test.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+          mongodb-version: ${{ matrix.mongodb-version }}
+
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Setup Conda and install python
+        uses: s-weigand/setup-conda@v1
+        with:
+          activate-conda: true
+          python-version: 3.11
```
Comment (henrikstranneheim) [inline .github/workflows/integration_test.yml:23]: Will you update the software to this version as well?

File: mutacc/mutaccDB/db_adapter.py [hunk 2]
```diff
@@ -68,20 +68,10 @@ def add_case(self, case):
...
-
-            Args:
-                case_id(str): name of case
-            Returns:
-                exists(bool): True if exists, else False
-        """
-
-        if self.cases_collection.find({"case_id": case_id}).count() > 0:
-            return True
+    def case_exists(self, case_id:str) ->  bool:
```
Comment (henrikstranneheim) [inline mutacc/mutaccDB/db_adapter.py:71]:
```suggestion
    def case_exists(self, case_id: str) ->  bool:
```

**Review by henrikstranneheim — APPROVED** [review]
👍

File: .github/workflows/integration_test.yml (hunk 1 above)
Comment (northwestwitch) [reply .github/workflows/integration_test.yml:23]: You mean the instructions or the Dockerfile?

Comment (henrikstranneheim) [reply .github/workflows/integration_test.yml:23]: Yes, so the actions and Dockerfile align

Comment (northwestwitch) [reply .github/workflows/integration_test.yml:23]: Ah OK, good idea, but I'll do it in another PR if it's OK

Comment (henrikstranneheim) [reply .github/workflows/integration_test.yml:23]: Perfect 👍

**Pull request merged**